server/consumer.go

```diff
@@ -2953,18 +3031,28 @@ func (o *consumer) setInitialPendingAndStart() {
 		}
 	} else {
 		// Here we are filtered.
-		ss := mset.store.FilteredState(o.sseq, o.cfg.FilterSubject)
-		if ss.Msgs > 0 {
+		dp := o.cfg.DeliverPolicy
+		if dp == DeliverLastPerSubject && o.hasSkipListPending() && o.sseq < o.lss.resume {
+			if o.lss != nil {
```

Comment:
i tink `o.hasSkipListPending()` check above implies o.lss will be non nil

Reply:
Upper clause has an additional conditional so I need to check again.

```go
if dp == DeliverLastPerSubject && o.hasSkipListPending() && o.sseq < o.lss.resume {
```